### DISCUSSION TITLE
Why do I read models with the Companion without connection to the radio

### DISCUSSION THREAD

**Discussion by luzrael** [EdgeTX/edgetx#2930] [Q&A]
Good evening.

Since I installed the first versions of EdgeTx I have had a lot of problems when reading/writing models with the Companion for EdgeTx.
So, I thought it was an EdgeTx problem.

Tonight I started the computer without having the radio connected (by forgetfulness), and I started the companion.

Without realizing that I had the radio disconnected, I see that the icon to read the models and settings of the radio was enabled, so I press it.

After waiting a couple of seconds, I get the file document1.etx with my last models/changes.

At that moment, I see that the radio is turned off, and not connected to the computer.
The radio was not connected to the computer, but the Companion "had read" the models that I was editing a couple of days ago!!!

So the question is.
        A) What models do you read?
	B) Why does it not detect that the radio is not connected?
	C) How do I fix this problem so that the Companion detects when the radio is connected.
	D) How to know that the Companion has correctly read/write the configuration and models from/to the radio.
	
Thanks a lot.

**Comment by pfeerick** [comment]
Under certain circumstances Companion can be tricked, as it is looking for a FAT filesystem which contains a `RADIO` folder... which is why you can remove the SD card and put it in a card reader. If one of your hard drives/USB drive, etc has a FAT filesystem and a RADIO folder, Companion will think that is the radio SD card. So you will most likely find if you check your drives that there is a RADIO and MODELS folder there, which is the culprit. 

The "Read" button is always enabled - it does not indicate presence of the radio or not.

**Reply by luzrael** [reply]
Thank you very much Peter Feerick!!!!

Without your help I would never have been able to use the Companion.

As well as driving me crazy to find out which drive I was leaving everything on 🥴🥴

It turns out that I had inserted a microSD in the port of the computer with the last backup of the radio. 

That backup was where I was getting the information from and that is why I never started from scratch and kept almost everything 🤦🤦🤦

Thank you very, very much 🙏🙏🙏
